component/api/src/main/java/org/exoplatform/social/core/storage/api/SpaceStorage.java

```diff
@@ -717,6 +717,29 @@ default List<String> getMemberSpaceIds(String identityId, int offset, int limit)
     return getMemberRoleSpaceIdentityIds(identityId, offset, limit);
   }
 
+  /**
+   * Retrieves the list of Space {@link Identity} technical identifiers
+   *
+   * @param userIdentityId user {@link Identity} technical identifier
+   * @param status {@link String} equals to MEMBER, MANAGER, REDACTOR, PENDING,
+   *          INVITED, IGNORED
+   * @param offset The starting point
+   * @param limit The limitation of returned results
+   * @return {@link List} of space {@link Identity} technical identifiers
+   */
+  List<String> getSpaceIdentityIdsByUserRole(String userIdentityId, String status, int offset, int limit);
+
+  /**
+   * Retrieves the list of favorite Space {@link Identity} technical identifiers
+   *
+   * @param userIdentityId user {@link Identity} technical identifier
+   * @param spaceFilter
+   * @param offset The starting point
+   * @param limit The limitation of returned results
+   * @return {@link List} of space {@link Identity} technical identifiers
+   */
+  List<String> getFavoriteSpaceIdentityIds(String userIdentityId, SpaceFilter spaceFilter, int offset, int limit);
```

Comment:
Use `default` implementation to avoid compilation errors  when changing existing API ([DEVSTD-23](https://github.com/Meeds-io/meeds/blob/develop/CODING_GUIDELINES.md#devstd-23))

```suggestion
  default List<String> getFavoriteSpaceIdentityIds(String userIdentityId, SpaceFilter spaceFilter, int offset, int limit) {
    throw new UnsupportedOperationException();
  }
```